secboot/secboot_tpm.go

```diff
@@ -502,27 +563,28 @@ func ResealKeys(params *ResealKeysParams) error {
 		return err
 	}
 
+	// FIXME: load primary key from keyring when available
 	authKey, err := os.ReadFile(params.TPMPolicyAuthKeyFile)
 	if err != nil {
 		return fmt.Errorf("cannot read the policy auth key file %s: %w", params.TPMPolicyAuthKeyFile, err)
 	}
 
 	keyDatas := make([]*sb.KeyData, 0, numSealedKeyObjects)
 	sealedKeyObjects := make([]*sb_tpm2.SealedKeyObject, 0, numSealedKeyObjects)
-	for _, keyfile := range params.KeyFiles {
-		loadedKey := &defaultKeyLoader{}
-		const hintExpectFDEHook = false
-		err := readKeyFile(keyfile, loadedKey, hintExpectFDEHook)
+	writers := make([]sb.KeyDataWriter, 0, numSealedKeyObjects)
+	for _, key := range params.Keys {
```

Comment:
hm one thing that isn't clear to me, is a case where one key is from a token a other is from a key file even possible? or if the first key is read from a token, then we're expecting all subsequent keys to come from the tokens as well?

Reply:
That could happen in theory if we had an incomplete reprovision which switches from files to tokens.

Reply:
Though we would not reseal in that case.

Reply:
Anyway, at this point, it does not matter if we can have a mix or not. That does not change our code. What we cannot mix is old format with new format.